### ISSUE TITLE
String parsing in phylanx is broken

### ISSUE THREAD

**Issue by stevenrbrandt** [STEllAR-GROUP/phylanx#1106]
This code, which only sets a string, fails

```
from phylanx import Phylanx

@Phylanx
def foo():
    s = "defined(hello,cout(\"hello\"))"

foo()
```
Results in the following error
```
$ python3 $PWD/v.py
--------------------------------------------------------------------------
[[25195,1],0]: A high-performance Open MPI point-to-point messaging module
was unable to find any relevant network interfaces:

Module: OpenFabrics (openib)
  Host: 468c33cbe4d5

Another transport will be used instead, although this may result in
lower performance.
--------------------------------------------------------------------------
PAPI lmsensors component found...
PAPI RAPL component found...
No RAPL events found.
Can't open fd for cpu0: Permission denied.
Traceback (most recent call last):
  File "/home/jovyan/.local/lib/python3.6/site-packages/phylanx-0.0.1-py3.6-linux-x86_64.egg/phylanx/ast/physl.py", line 386, in _compile_or_load
    self.__ast__ = phylanx.ast.generate_ast(self.__src__)
RuntimeError: Error! Expecting ")" line 1:
define$4$0(foo$4$0, lambda$4$0(define$5$4(s$5$4, "defined(hello,cout("hello"))")))
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
Error! Expecting ")" line 1:
define$4$0(foo$4$0, lambda$4$0(define$5$4(s$5$4, "defined(hello,cout("hello"))")))
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
Error! Expecting ")" line 1:
define$4$0(foo$4$0, lambda$4$0(define$5$4(s$5$4, "defined(hello,cout("hello"))")))
~~~~~~~~~~~~~~~~~~^
Error! Incomplete parse: line 1:
define$4$0(foo$4$0, lambda$4$0(define$5$4(s$5$4, "defined(hello,cout("hello"))")))
^
: HPX(bad_parameter)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/jovyan/phylanx/v.py", line 7, in <module>
    foo()
  File "/home/jovyan/.local/lib/python3.6/site-packages/phylanx-0.0.1-py3.6-linux-x86_64.egg/phylanx/ast/transducer.py", line 162, in __call__
    result = self.backend.call(*mapped_args, **mapped_kwargs)
  File "/home/jovyan/.local/lib/python3.6/site-packages/phylanx-0.0.1-py3.6-linux-x86_64.egg/phylanx/ast/physl.py", line 548, in call
    self._ensure_global_state()
  File "/home/jovyan/.local/lib/python3.6/site-packages/phylanx-0.0.1-py3.6-linux-x86_64.egg/phylanx/ast/physl.py", line 421, in _ensure_global_state
    PhySLFunction.compile()
  File "/home/jovyan/.local/lib/python3.6/site-packages/phylanx-0.0.1-py3.6-linux-x86_64.egg/phylanx/ast/physl.py", line 348, in compile
    func.compile_function()
  File "/home/jovyan/.local/lib/python3.6/site-packages/phylanx-0.0.1-py3.6-linux-x86_64.egg/phylanx/ast/physl.py", line 342, in compile_function
    self.physl._ensure_is_compiled()
  File "/home/jovyan/.local/lib/python3.6/site-packages/phylanx-0.0.1-py3.6-linux-x86_64.egg/phylanx/ast/physl.py", line 434, in _ensure_is_compiled
    self._compile_or_load()
  File "/home/jovyan/.local/lib/python3.6/site-packages/phylanx-0.0.1-py3.6-linux-x86_64.egg/phylanx/ast/physl.py", line 400, in _compile_or_load
    self.__ast__ = phylanx.ast.generate_ast(self.__src__)
RuntimeError: Error! Expecting ")" line 1:
define$4$0(foo$4$0, lambda$4$0(store$5$4(s$5$4, "defined(hello,cout("hello"))")))
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
Error! Expecting ")" line 1:
define$4$0(foo$4$0, lambda$4$0(store$5$4(s$5$4, "defined(hello,cout("hello"))")))
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
Error! Expecting ")" line 1:
define$4$0(foo$4$0, lambda$4$0(store$5$4(s$5$4, "defined(hello,cout("hello"))")))
~~~~~~~~~~~~~~~~~~^
Error! Incomplete parse: line 1:
define$4$0(foo$4$0, lambda$4$0(store$5$4(s$5$4, "defined(hello,cout("hello"))")))
^
: HPX(bad_parameter)
```

**Comment by stevenrbrandt** [comment]
Note that this works if one uses single quotes rather than double quotes inside the string.

**Comment by hkaiser** [comment]
> Note that this works if one uses single quotes rather than double quotes inside the string.

Sure, makes sense. We need to check in physl.py when generating the code if a string holds `\"` or `\'` characters and adapt accordingly. @rtohid any ideas?

**Comment by rtohid** [comment]
@hkaiser, Python AST automatically removes the escape chars and replace them with `'` or `"`- whichever applicable, for example the `ast.Str` node in the following code:

```py
def foo():
    s = "defined(hello,cout(\"hello\"))"

import ast
import astpretty
import inspect

_ast = ast.parse(inspect.getsource(foo))

astpretty.pprint(_ast)
```

looks like (notice that the outer quotation is now sinlgle):
`value=Str(lineno=2, col_offset=8, s='defined(hello, cout("hello"))')`

@stevenrbrandt. you may want to check the outer quotation and if it is `'`, swap all single quotation with double and vice versa before [returning](https://github.com/STEllAR-GROUP/phylanx/blob/ee2487275ad5b653ad0e5fae1e882a0f4404c6a5/python/phylanx/ast/physl.py#L1237).

**Comment by hkaiser** [comment]
> @stevenrbrandt. you may want to check the outer quotation and if it is ', swap all single quotation with double and vice versa before returning.

Should be easy enough to fix there. Any takers?

**Comment by stevenrbrandt** [comment]
@hkaiser @rtohid https://github.com/STEllAR-GROUP/phylanx/pull/1108

**Comment by hkaiser** [comment]
This has been fixed by merging #1108
